tests/unit/test_core/test_cache.py

```
@@ -0,0 +1,588 @@
+# Helper class for async iteration
+class AsyncIteratorMock:
+    """Mock for async iterators like scan_iter."""
+
+    def __init__(self, items: list[Any]) -> None:
+        self.items = items
+        self.index = 0
+
+    def __aiter__(self) -> AsyncIteratorMock:
+        return self
+
+    async def __anext__(self) -> Any:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `__anext__`

(ANN401)